### DISCUSSION TITLE
How do I access a certain API function using the CLI version?

### DISCUSSION THREAD

**Discussion by DaforLynx** [FluidSynth/fluidsynth#1278] [Q&A] [answered]
Hi, for reasons, I need to use [this API setting](https://www.fluidsynth.org/api/group__synthesis__params.html#ga4a2efef77b267500dd9c19c0dc9e4633) to change the interpolation method of FluidSynth's sampling (I'm assuming it changes the interpolation of each voice rather than the overall output). Is there any way I can do this? The closest I can find is using `-o audio.oboe.sample-rate-conversion-quality` but of course I'm not using Oboe and it probably doesn't do what I want it to do anyway.

**Comment by DaforLynx** [comment]
Update: after much trial and error I managed to compile it using the desired interp method
Would still be nice to be able to access it in the CLI

**Comment by derselbst** [comment, marked as answer]
Fluidsynth's shell provides the commands `interp` and `interpc` for this purpose, see `help general`. Put these commands in a text file and pass it to fluidsynth using the `-f` option, see [UserManual](https://github.com/FluidSynth/fluidsynth/wiki/UserManual).

**Reply by DaforLynx** [reply]
Either it doesn't work, or the User Manual is very unhelpful. I passed a config txt using `-f` containing just `interp num 0` 
[fluid_config.txt](https://github.com/FluidSynth/fluidsynth/files/12675765/fluid_config.txt) and it didn't do anything. Adding a return after it didn't do anything. I could confirm this by inputting the command manually and noting the difference. Putting quotes around the `"interp num 0"` gave `unknown command: interp num 0 (try help)` as an error, and it didn't go through.
Current version is 2.3.3.

**Reply by derselbst** [reply]
> interp num 0

You are supposed to replace `num` with the number corresponding for the interpolation, i.e. `interp 0`.

**Reply by DaforLynx** [reply]
Weird how the command `interp num 0` actually worked on the command line itself, then. I didn't think to replace num. Anyway, thanks for the help here. If it works I'll mark it as answered

**Reply by DaforLynx** [reply]
@derselbst Still doesn't work. If I choose `interp -1` it throws a bad value error like expected. I can't tell for sure whether other numbers work because the difference isn't as obvious. I also found the reason `interp num 0` works is because it defaults  to `0` when it gets an unknown quantity. But only during runtime in the command line! And a bad value like `-1` will still throw a bad value error. I don't know what's going on. It seems the `interp` command does not work when read from a file using `-f`.

**Reply by mawe42** [reply]
The interp command has two forms: with a single argument for all channels, or with two arguments for a specific channel. To set an interp value for all channels, use `interp 0` or some other valid interp value. See https://www.fluidsynth.org/api/group__synthesis__params.html for valid values.

**Reply by DaforLynx** [reply]
You are not understanding. I know how the command works. It just does not work when used _correctly_ with `-f`, despite seeming to know when it is used _incorrectly_ with the same option. And according to the manual, the command to set interp per channel is `interpc`.

**Reply by DaforLynx** [reply]
I think this has strayed into "issue" territory so unless it turns out I'm missing something hugely obvious, I'll make an issue instead.

**Reply by mawe42** [reply]
Which version of FluidSynth are you using to test this?

**Reply by mawe42** [reply]
And how are you testing the effect of the interp command? Playing a MIDI file?

**Reply by DaforLynx** [reply]
I am using 2.3.3 as stated previously. I've also confirmed it using a self-compiled latest version.

**Reply by DaforLynx** [reply]
Yes. I play a MIDI file using an sf2 and use the option `-f "path\to\txt"`, with said txt only containing `interp 0`.

**Reply by mawe42** [reply]
Please check if your MIDI files contains a system reset message (which would also reset any interpolation settings previously set via the config file). Using `-v` on the command line will output all midi messages to the console.

**Comment by DaforLynx** [comment]
That appears to be the case.
![image](https://github.com/FluidSynth/fluidsynth/assets/13303368/2c51c58f-e8c9-4eba-86b3-49aef5f368ff)
 Since it doesn't appear there's any way I can change this via my preferred MIDI editor, I guess I'll just stick to using my self-compiled version. Is there a way I could justify suggesting the interp settings *not* change on a system reset message?

**Reply by mawe42** [reply]
The [MIDI Spec](https://www.midi.org/specifications-old/item/table-1-summary-of-midi-message) says the following about system reset messages: "Reset all receivers in the system to power-up status. This should be used sparingly, preferably under manual control. In particular, it should not be sent on power-up."

So I in my opinion, FluidSynths behaviour is correct here: changing the interpolation method is a run-time setting that gets reset to power-up status on system reset. And to be honest, I think it's quite unusual for a MIDI file to contain a system reset.

**Reply by DaforLynx** [reply]
Alright, that's that, then. It's a result of a translation process (I'm using VGMTrans to rip some MIDIs, and...well, it's irrelevant) and I just can't seem to find a way to get rid of those messages. It's alright, though. Thanks for the help in diagnosis. And I'll mark derslbt's original comment as the answer.
